content/spin/v2/testing-apps.md

```diff
@@ -0,0 +1,196 @@
+
+There is currently first-class support for Rust, but any language with support for writing WebAssembly components can be used as long as the `fermyon:spin-test/test` world is targeted. You can find the definition of this world [here](https://github.com/fermyon/spin-test/blob/4dcaf79c10fc29a8da2750bdaa383b5869db1715/host-wit/world.wit#L13-L16).
+
+Here’s an example of a test written in Rust using the [Spin Test Rust SDK](https://github.com/fermyon/spin-test/tree/main/crates/spin-test-sdk) that tests to ensure that the Spin app responds properly when the key-value store has a certain key already set.
+
+Open the `my-component/tests/src/lib.rs` file and fill it with the following content:
+
+```rust
+use spin_test_sdk::{
+    bindings::{
+        // fermyon::spin_test_virt::{http_handler, key_value},
+        wasi::http,
+    },
+    spin_test,
+};
+
+#[spin_test]
+fn it_works() {
+    make_request();
+}
+
+fn make_request() {
+    // Perform the request
+    let request = http::types::OutgoingRequest::new(http::types::Headers::new());
+    request.set_path_with_query(Some("/?user_id=123")).unwrap();
+    let response = spin_test_sdk::perform_request(request);
+
+    // Assert response status and body
+    assert_eq!(response.status(), 200);
+}
+```
+
+The following points are intended to unpack the above example for your understanding:
+
+- Each function marked with `#[spin_test]` will be run as a separate test.
+- Each test can perform any setup to their environment by using the APIs available in `spin_test_sdk::bindings::fermyon::spin_test_virt`.
+- Requests are made to the Spin application using the `spin_test_sdk::perform_request` function.
+- After requests are made, you can use the APIs in `spin_test_sdk::bindings::fermyon::spin_test_virt` to check how the request has changed the state of various resources (such as the key/value store) and make assertions that things changed in the way you expected.
+
+The test above will run inside of WebAssembly. Calls, such as Key Value storage and retrieval, never actually leave the WebAssembly sandbox. This means your tests are quick and reproducible as you don’t need to rely on running an actual web server, and you don’t need to ensure any of your app’s dependencies are running. Everything your app interacts with is mocked for you.
+
+<!-- markdownlint-disable-next-line titlecase-rule -->
+## Configure `spin-test`
+
+Before you can run the test, you'll need to tell `spin-test` where your test lives and how to build it. You do this from inside our app’s manifest (the `spin.toml` file). We change back up into our application's root directory:
+
+<!-- @selectiveCpy -->
+
+```bash
+$ cd ..
+```
+
+Then we edit the application's manifest (the `spin.toml` file) as shown below:
+
+<!-- @selectiveCpy -->
+
+```toml
+[component.my-component.tool.spin-test]
+source = "tests/target/wasm32-wasi/release/tests.wasm"
+build = "cargo component build --release"
+dir = "tests"
```

Comment:
Check with Ryan - I think this is imminently to be renamed `workdir` for consistency with the Spin `build` section.